File: takeme-webapp/src/main/webapp/app/shared/model/child-dto.ts
import { Moment } from "moment";

export class ChildDTO {
  name: string;
  nifCountry: string;
  nifNumber: string;
  dateOfBirth: Moment;
  photo: string;

  constructor(name: string, nifCountry: string, nifNumber: string, dateOfBirth: Moment, photo: string) {
      this.name = name;
      this.nifCountry = nifCountry;
      this.nifNumber = nifNumber;
      this.dateOfBirth = dateOfBirth;
      this.photo = photo;
  }
}

File: takeme-webapp/src/main/java/pt/famility/backoffice/service/dto/ChildDTO.java
package pt.famility.backoffice.service.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import pt.famility.backoffice.domain.enumeration.StatusType;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
public class ChildDTO {

    private Long id;

    @NotBlank
    private String name;

    @NotBlank
    private String nifNumber;

    @NotNull
    private LocalDate dateOfBirth;

    private Boolean famility;

    private StatusType statusType;

    private String nifCountry;

    private String photo;

    private Long photoId;

    private List<ChildSubscriptionChildViewDTO> activeSubscriptions = new ArrayList<>();

}
